tools/topology/sof-tgl-max98373-rt5682.m4

```diff
@@ -158,6 +158,22 @@ PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
 	1000, 0, 0,
 	48000, 48000, 48000)
 
+# Deep-buffer playback pipeline 15 on PCM 5 using max 2 channels of s32le.
+# Schedule 960 frames per 20000us deadline on core 0 with priority 0
+ifdef(`INCLUDE_DEEP_BUFFER',
+`PIPELINE_PCM_ADD(sof/pipe-pcm-media.m4,
+	15, 15, 2, s32le,
+	20000, 0, 0,
+	48000, 48000, 48000)
+SectionGraph."media-pipeline" {
+	index "0"
+
+	lines [
+		# media 0
+		dapm(PIPELINE_MIXER_2, PIPELINE_SOURCE_15)
```

Comment:
No, let's not do this.
We should *NEVER* use a mixer inside a pipeline, we have to split between host pipelines (low-latency and deep-buffer) and have a dai pipeline that starts with a mixer.

If we use a deep-buffer, let's do it the right way.

Reply:
@plbossart do you mind if I switch all the topologies that use the pipe-low-latency-playback pipeline to have the mixer moved to the DAI pipeline as well?

Reply:
I don't mind, i've been preaching for this since we started but we have to be realistic on validation effort. Let's start with e.g. TGL topologies that are checked in CI/daily tests.

Reply:
What's the plan to support low latency sounds during deepbuffer if we don't have a mixer ? e.g. we could have a 100ms to 2s host buffer and a 20ms .. 100ms DSP buffer, so worst case we could be looking at 100+ ms to hear a keypress or other low latency sound.

Reply:
we will have a mixer, just not in the middle of a pipeline.

Reply:
@lgirdwood see the diagrams at https://docs.microsoft.com/en-us/windows-hardware/drivers/audio/audio-processing-object-architecture, this is the type of hiearchy we need to have to properly support post-processing. The PCM-to-dai pipeline is a dead-end and doesn't scale to multi-level mixing.

Reply:
![audio-apo-software-effects-summary](https://user-images.githubusercontent.com/1149035/119891310-6fb47480-befe-11eb-9b1b-dbe52548e75f.png)

Reply:
ok, so just before the DAI  or at the host end ? At the DAI end would give the lowest latency, but we may also need a volume control after the mixer and also volumes for each mixer source ? If we use volumes then it puts the mixer back in the middle, so it may need to be some sort of compromise ?

Reply:
the DAI needs to operate at the lowest latency, e.g. 1ms. We need volume control before and after any mixer.

I don't know what you meant by 'mixer in the middle', in the example above we would have one pipeline for each PCM + SFX, two pipelines for mixing+MFX and the final pipeline for Raw+mixing+EFX.

Reply:
Sorry, I'm confused too, the bottom APO diagram shows the DSP with 2 mixers (1 mixers 2 PCMs) the second mixer appears to mix nothing where audio is rendered on the speaker., Are you suggesting we have a similar N PCMs with a mixer in the DSP ?

Reply:
@lgirdwood I was only describing what takes place in the 'Audio Engine'. We need a similar hierarchical set of mixers in the DSP managed by SOF.

Reply:
Now, I've got you @plbossart - @ranj063 and I had a chat and aligned that the mixer is on the DAI end of the DSP pipeline as the final stage (with possible a main volume) prior to the DAI. Thanks.